### ISSUE TITLE
How to bind my model with an action to a DitFloView PartialView

### ISSUE THREAD

**Issue by lele7** [leekelleher/umbraco-ditto#193]
I've got a binding error in the DitFloView if I call an action on my DitFloController.

`    public PartialViewResult DoSomething()
    {
      var typedView = new typedModel();
      return PartialView(typedView);
    }
`

On the view I call it as below:
`@Html.Action("DoSomething")`

Ther error occurs on this line in the DitFloView.cs
`base.SetViewData(newViewData);`

thanks for your help.

**Comment by leekelleher** [comment]
Hi @lele7, which version of Ditto are you using?

The DitFlo feature was merged into the main Ditto project in v0.10 (the most recent version) and renamed to DittoView.

**Comment by lele7** [comment]
Hi leekelleher

I'm using the version 0.9 with the (latest) DitFlo project... Matt told me already that you guys merged it together... but he means it's more or less the same than the latest DitFlo project

**Comment by mattbrailsford** [comment]
Could you show me your code in DoSomething.cshtml partial?

**Comment by lele7** [comment]
```
@inherits Our.Umbraco.DitFlo.Web.Mvc.DitFloView<typedModel>

<h2>@Model.View.Title</h2>

@foreach (var itm in Model.View.Items)
{
<p>@itm.Subtitle</p>
}

```

**Comment by leekelleher** [comment]
@lele7 Does the `typedModel` class have a namespace? It looks like your partial is missing a namespace reference.

**Comment by lele7** [comment]
yup sorry sure the namespace is:
`Company.Framework.ViewModels.TypedModel`

**Comment by leekelleher** [comment]
So if you change the partial directive to...

```csharp
@inherits Our.Umbraco.DitFlo.Web.Mvc.DitFloView<Company.Framework.ViewModels.TypedModel>
```
... that should work.

**Comment by lele7** [comment]
Sadly not. I have the namespace in the directive...

**Comment by leekelleher** [comment]
Ah ok, Just that it was missing in the original code snippet. Hmmm, not really got much to go on here - sorry.

If your controller action is already returning the correct object type, then you don't need to use DitFlowView, I believe UmbracoViewPage be fine. (Unless I'm missing something?)

**Comment by lele7** [comment]
ok sorry I forgot it to add first. ok I'll try the UmbracoViewPage.

**Comment by mattbrailsford** [comment]
Using DittoView in your partial should be totally valid. Leave it with me and I'll see if i can replicate.

**Comment by lele7** [comment]
ok thanks

**Comment by mattbrailsford** [comment]
Can you show me the whole code for your controller?

**Comment by mattbrailsford** [comment]
This is what I have setup (using Ditto 0.10) and it all works

Controllers/TestPartialController.cs

    using System.Web.Mvc;
    using DittoDemo.ViewModels;
    using Our.Umbraco.Ditto;

    namespace DittoDemo.Controllers
    {
        public class TestPartialController : DittoController
        {
            [ChildActionOnly]
            public PartialViewResult DoSomething()
            {
                var typedView = new TestPartialViewModel();

                typedView.MyProp = "HELLO WORLD";

                return CurrentPartialView(typedView);
            }
        }
    }

ViewsModels/TestPartialViewModel.cs

    namespace DittoDemo.ViewModels
    {
        public class TestPartialViewModel
        {
            public string MyProp { get; set; }
        }
    }

Views/Partials/DoSomething.cshtml

    @inherits DittoView<DittoDemo.ViewModels.TestPartialViewModel>
    <h1>@Model.View.MyProp</h1>

Views/Layout.cshtml

    ...
    @Html.Action("DoSomething", "TestPartial")
    ...

**Comment by lele7** [comment]
Thanks for your quick help. I will check my code and comeback.

**Comment by lele7** [comment]
hmm I have a strange issue... I've updated to the latest ditto version 0.10. I've got this error if I open the site: The resource cannot be found.  If I removing the inheritance the site is working without contoller... so I think I have a routing issue...

If I try it with the SurfaceController inheritance the site is working too...

Do you have any idea why? Could it be an Umbraco version issue?
I'm using 7.5.2

**Comment by mattbrailsford** [comment]
Strange, minimum requirement it 7.3.2 so shouldn't be any issues with 7.5.2. Can you send a fully copy of the stack trace?

**Comment by lele7** [comment]
Server Error in '/' Application.

The resource cannot be found.

Description: HTTP 404. The resource you are looking for (or one of its dependencies) could have been removed, had its name changed, or is temporarily unavailable.  Please review the following URL and make sure that it is spelled correctly. 

Requested URL: /fr/services/etudes-comparatives

Version Information: Microsoft .NET Framework Version:4.0.30319; ASP.NET Version:4.6.1055.0

**Comment by lele7** [comment]
could it be that somewhere is missing the host part? in my opinion should it be the requested URL with hostname...

**Comment by leekelleher** [comment]
Hi @lele7,

Since this isn't about a specific bug (or a bug that we can reproduce ourselves), it seems more like a general implementation issue.  Can you start a thread in the forum (over on Our Umbraco) please?

<https://our.umbraco.org/projects/developer-tools/ditto/ditto-feedback/>

We prefer to keep the GitHub issues to specific bugs and feature requests. (This is outlined in our [contribution guidelines](https://github.com/leekelleher/umbraco-ditto/blob/develop/CONTRIBUTING.md).)

We'll try to help where we can over on Our Umbraco.  Then if a specific bug with Ditto has been found, it can be raised back on here and we'll attempt to resolve it.

Thanks!

**Comment by lele7** [comment]
Sure, thank you guys for your help!

**Comment by lele7** [comment]
Hey guys

I think I find the issue... if I use the DittoView with an Ajax action call to update a PartialView the SetDataView on DittoView throws an null ref exception... 

Umbraco had an issue in 7.4.0 (http://issues.umbraco.org/issue/U4-7950) which should fixed upper 7.4.1.

If I change the PartialView inheritance to UmbracoViewPage all works as expected.
